### PR TITLE
ddl: refine history job code

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

refine history job code
- rename function name
- use `iterator` implement all the history job relate function

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: meta/meta.go [hunk 1]
```diff
@@ -1016,36 +1015,11 @@ func (m *Meta) GetHistoryDDLJob(id int64) (*model.Job, error) {
...
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	jobs, err := decodeJob(pairs)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	// sort job.
-	sorter := &jobsSorter{jobs: jobs}
-	sort.Sort(sorter)
```
Comment (hawkingrei) [inline meta/meta.go:1031]: In the old code. jobs will be sorted by the ID. But I don't know the NewHashReverseIter whether to sort them.

Comment (xiongjiwei) [reply meta/meta.go:1031]: it will, but it is in reverse order, I checked the code, and the order does not matter

**Comment by MimeLyc** [comment]
update

File: ddl/ddl.go [hunk 2]
```diff
@@ -1193,7 +1194,22 @@ func IterAllDDLJobs(txn kv.Transaction, finishFn func([]*model.Job) (bool, error
...
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	allJobs := make([]*model.Job, 0, batchNumHistoryJobs)
+	for {
+		jobs, err := iterator.GetLastJobs(batchNumHistoryJobs, nil)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if len(jobs) == 0 {
```
Comment (Defined2014) [inline ddl/ddl.go:1207]: maybe `len(jobs) < batchNumHistoryJobs` is enough

File: meta/meta_test.go [hunk 3]
```diff
@@ -564,7 +565,7 @@ func TestDDL(t *testing.T) {
 			historyJob2.Args = append(job.Args, arg)
 			err = m.AddHistoryDDLJob(historyJob2, false)
 			require.NoError(t, err)
-			all, err := m.GetAllHistoryDDLJobs()
+			all, err := ddl.GetAllHistoryDDLJobs(m)
```
Comment (Defined2014) [inline meta/meta_test.go:568]: Seems some problems with UT. 
https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/57424/tests

File: ddl/ddl.go (hunk 2 above)
Comment (xiongjiwei) [reply ddl/ddl.go]: done

File: meta/meta.go (hunk 1 above)
Comment (xiongjiwei) [reply meta/meta.go:1031]: it is the same as before now.

**Review by Defined2014 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]
I think we can use iterator to optimize the get ddl history HTTP api, like paging.

**Comment by xiongjiwei** [comment]
> I think we can use iterator to optimize the get ddl history HTTP api, like paging.

that is another job

**Pull request merged**